src/unix/pipe.c

```diff
@@ -361,3 +361,21 @@ int do_pipe2(int fds[2], int flags)
     pipe_release(pipe);
     return -ENOMEM;
 }
+
+int pipe_set_capacity(fdesc f, int capacity)
+{
+    pipe_file pf = (pipe_file)f;
+    pipe p = pf->pipe;
+    if (capacity < INITIAL_PIPE_DATA_SIZE)
```

Comment:
Any idea where this INITIAL_PIPE_DATA_SIZE value of 100 comes from? Seems from pipe(7) that the lower limit should be PAGESIZE. Also, do we want to place an upper bound here? (man page claims default for /proc/sys/fs/pipe-max-size is 1MB).

Reply:
The INITIAL_PIPE_DATA_SIZE value of 100 has been there since the initial pipe implementation, I would guess it's just a made up number. I'm OK with setting the lower limit to PAGESIZE, but as far as an upper limit is concerned, the man pages say that the /proc/sys/fs/pipe-max-size limit applies only to unprivileged processes, while a privileged process (CAP_SYS_RESOURCE) can override the limit. So there is no really upper bound for privileged processes, and I think the process running on a unikernel should be as privileged as it can get :-)